protocol/messenger_mailserver_cycle.go

```diff
@@ -650,3 +651,18 @@ func (m *Messenger) getPinnedMailserver() (*mailservers.Mailserver, error) {
 
 	return nil, nil
 }
+
+func (m *Messenger) EmitMailserverAvailable() {
+	for _, s := range m.mailserverCycle.availabilitySubscriptions {
+		s <- struct{}{}
+		close(s)
+		l := len(m.mailserverCycle.availabilitySubscriptions)
+		m.mailserverCycle.availabilitySubscriptions = m.mailserverCycle.availabilitySubscriptions[:l-1]
+	}
+}
+
+func (m *Messenger) SubscribeMailserverAvailable() chan struct{} {
+	c := make(chan struct{})
+	m.mailserverCycle.availabilitySubscriptions = append(m.mailserverCycle.availabilitySubscriptions, c)
+	return c
+}
```

Comment:
@Samyoul I've updated this as per our discussions. `EmitMailserverAvailable` now merely emits a signal and closes directly without waiting for a cancel signal from a second channel